ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -132,29 +139,78 @@ synchronized void updateIndex(long i) {
     }
   }
 
+  static class Queues implements ToIntFunction<String> {
+    private final DataBlockingQueue<Task> taskQueue;
+    private final WriteLogTasks writeTasks = new WriteLogTasks();
+
+    private final ByteBuffer writeBuffer;
+    private final Supplier<byte[]> sharedBuffer;
+
+    Queues(String name, RaftProperties properties) {
+      final SizeInBytes queueByteLimit = RaftServerConfigKeys.Log.queueByteLimit(properties);
+      final int queueElementLimit = RaftServerConfigKeys.Log.queueElementLimit(properties);
+      this.taskQueue = new DataBlockingQueue<>(name, queueByteLimit, queueElementLimit, Task::getSerializedSize);
+
+      final int bufferSize = RaftServerConfigKeys.Log.writeBufferSize(properties).getSizeInt();
+      this.writeBuffer = ByteBuffer.allocateDirect(bufferSize);
+      final int logEntryLimit = RaftServerConfigKeys.Log.Appender.bufferByteLimit(properties).getSizeInt();
+      // 4 bytes (serialized size) + logEntryLimit + 4 bytes (checksum)
+      this.sharedBuffer = () -> new byte[logEntryLimit + 8];
```

Comment:
@adoroszlai , The buffer is not shared so that TestStorageContainerManagerHA can pass.  If we use `MemoizedSupplier` as below, it will fail.
```java
      this.sharedBuffer = MemoizedSupplier.valueOf(() -> new byte[logEntryLimit + 8]);
```